### PR TITLE
Check for error condition before setting ERROR_MSG

### LINKED ISSUE #1696: DefaultLogHandler should follow OpenTracing spec
According to the [OpenTracing spec](https://github.com/opentracing/specification/blob/master/semantic_conventions.md#logged-application-level-errors), the presence of the `message` field in a log event does not necessarily indicate an error. However, the [DefaultLogHandler](https://github.com/DataDog/dd-trace-java/blob/58ba1f8/dd-java-agent/instrumentation/opentracing/src/main/java/datadog/trace/instrumentation/opentracing/DefaultLogHandler.java#L37-L44) sets the `ERROR_MSG` tag on the span if the `message` field is present. This causes otherwise successful requests that include a `message` to include and display an "error" message in APM
```json
{
  "error": {	
    "message": "whatever message"
  }
}
```

I believe the logger should only set `ERROR_MSG` to the value of `message` if `span.isError()` is true.

### PR DESCRIPTION
**Pull request by jfklingler**
Fixes #1696

### CODE REVIEW

File: dd-java-agent/instrumentation/opentracing/src/main/java/datadog/trace/instrumentation/opentracing/DefaultLogHandler.java [hunk 1]
```diff
@@ -34,11 +35,15 @@ public void log(final long timestampMicroseconds, final String event, final Agen
     log.debug("`log` method is not implemented. Provided log: {}", event);
   }
 
+  private boolean isErrorSpan(final Map<String, ?> map, final AgentSpan span) {
+    return (map.get(EVENT) instanceof String && map.get(EVENT).equals("error")) || span.isError();
```
Comment (richardstartin) [inline dd-java-agent/instrumentation/opentracing/src/main/java/datadog/trace/instrumentation/opentracing/DefaultLogHandler.java:39]: Can we assign `map.get(EVENT)` to a local and check `span.isError()` first please?

**Comment by jfklingler** [comment]
I added the logic changes, but I'm still working on a test.

**Comment by jfklingler** [comment]
I found another `DefaultLogHandler`. This one had tests nearby so that made those easier. I'm not quite sure if the two `DefaultLogHandler`s need to stay in sync, but I'm assuming so since they were identical.

**Comment by jfklingler** [comment]
FYI, I've run the `CustomLogManagerTest` that failed in `ci/circleci: test_13` a couple times and it passes locally. It says it was a timeout, but I wanted to make sure the timeout wasn't hiding a real error. If that indicates some other problem that isn't obvious, let me know and I'll take a look.

File: dd-trace-ot/src/test/groovy/datadog/opentracing/DefaultLogHandlerTest.groovy [hunk 2]
```diff
@@ -61,16 +61,80 @@ class DefaultLogHandlerTest extends DDSpecification {
...
+  }
+
+  def "handles correctly the message passed in the fields when the event is error"() {
+    setup:
+    final LogHandler underTest = new DefaultLogHandler()
+    final DDSpan span = tracer.buildSpan("op name").withServiceName("foo").start()
+    final String errorMessage = "errorMessage"
+    final Map<String, ?> fields = new HashMap<>()
+    fields.put(Fields.EVENT, "error")
+    fields.put(Fields.MESSAGE, errorMessage)
```
Comment (tylerbenson) [inline dd-trace-ot/src/test/groovy/datadog/opentracing/DefaultLogHandlerTest.groovy:122]: pro tip:

Groovy allows for inline declaration of maps: `[(Fields.EVENT): "error, (Fields.MESSAGE): errorMessage]`

Consider updating?

File: dd-trace-ot/src/test/groovy/datadog/opentracing/DefaultLogHandlerTest.groovy [hunk 3]
```diff
@@ -61,16 +61,80 @@ class DefaultLogHandlerTest extends DDSpecification {
...
+
+    then:
+    span.getTags().get(DDTags.ERROR_MSG) == errorMessage
+  }
+
+  def "handles correctly the message passed in the fields when called with timestamp when the span is error"() {
+    setup:
+    final LogHandler underTest = new DefaultLogHandler()
+    final DDSpan span = tracer.buildSpan("op name").withServiceName("foo").start()
+    final String errorMessage = "errorMessage"
```
Comment (tylerbenson) [inline dd-trace-ot/src/test/groovy/datadog/opentracing/DefaultLogHandlerTest.groovy:103]: I often like to put arbitrary test data like this in the `where:` block.

File: dd-trace-ot/src/test/groovy/datadog/opentracing/DefaultLogHandlerTest.groovy (hunk 2 above)
Comment (tylerbenson) [reply dd-trace-ot/src/test/groovy/datadog/opentracing/DefaultLogHandlerTest.groovy:122]: actually, I just realized this is also the case in the other test methods so never mind.

**Pull request merged**